### DISCUSSION TITLE
How to change page metadata ?

### DISCUSSION THREAD

**Discussion by Arwex007** [statiqdev/Discussions#154] [Q&A] [answered]
Hello ! I just want to whenever I click in a specific button, it modifies the ArchivePageSize metadata in the index page, to load more posts, how should I proceed ? Thanks!

**Comment by daveaglick** [comment, marked as answer]
Hi! There may be some confusion about how static generators work. Statiq generates HTML and other files that get sent to the web browser. At “runtime” (I.e. when the user is browsing the site), Statiq is completely out of the picture. Any dynamic client behavior needs to be implemented in JavaScript (and that’s not to say a static site can’t have lots of dynamic client behavior - just that it needs to be provided by client-side code). Because of that, mixing things like clicking buttons (a runtime behavior) with page metadata (a generation-time concept) doesn’t really work.

Does that make sense?

**Reply by Arwex007** [reply]
Oh, I see, but can I make a new page like the (Older, Newer) buttons, but with all the posts ? Without having the feel to refresh the page like it is ?

**Reply by daveaglick** [reply]
Sure! The trick is to make sure all the possibilities are generated beforehand. In this case, if you want a separate page that has all posts (or wherever other kind of content) you can use the archive feature without defining a front matter value for `ArchiveKey` (so it doesn’t group items) or `ArchivePageSize` (so it doesn’t paginate items). Then you can link to the generated single archive page from anywhere.

**Reply by Arwex007** [reply]
Do you have more detailed instructions, please? I'm not really fine with Statiq, and I just want to achieve that.

Thanks in advance

**Reply by daveaglick** [reply]
Sure - there’s a whole set of archives examples here: https://www.statiq.dev/guide/examples/archives

**Reply by Arwex007** [reply]
Okay well, I have created two pages, two are identical except the fact that the first one has only 5 on `ArchivePageSize` and the other has no limit. When I click on a button, I link to the other page. But the fact is we can clearly see the refresh. Is there any solution or not all ?

Thanks!

**Reply by Arwex007** [reply]
Furthermore, the URL endpoint, is not the first page as I expected, it's the second ; so it's like the user can clearly see that it's a copy of the original page rather than creating an “illusion” for the user.

**Reply by daveaglick** [reply]
That gets into client-side SPA-like magic and is pretty far outside the scope of Statiq. You'll likely need to make an XHR request to get the content of the "second" page and then do some DOM manipulation to swap the current content in a given container div with the new content you fetched. Something like https://github.com/turbolinks/turbolinks or https://github.com/Dogstudio/highway could help.
